bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/WebThingHandler.java

```diff
@@ -0,0 +1,302 @@
+
+                    // update the Thing structure based on the WebThing description
+                    thingStructureChanged(webThing);
+
+                    // link the Thing's channels with the WebThing properties to forward properties/item updates
+                    establishWebThingChannelLinks(webThing);
+
+                    lastReconnect.set(Instant.now());
+                    updateStatus(ThingStatus.ONLINE);
+                    return true;
+                }
+            } catch (IOException e) {
+                var msg = e.getMessage();
+                if (msg == null) {
+                    msg = "";
+                }
+                onError(msg);
+            }
+        }
+        return false;
+    }
+
+    public void onError(String reason) {
+        var wasConnectedBefore = isOnline();
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, reason);
+
+        // close the WebThing connection. If the handler is still active, the WebThing connection
+        // will be re-established within the periodically watchdog task
+        webThingConnectionRef.getAndSet(Optional.empty()).ifPresent(ConsumedThing::close);
+
+        if (wasConnectedBefore) { // to reduce log messages, just log in case of connection state changed
+            logger.debug("WebThing {} disconnected {}. Try reconnect (each {} sec)", getWebThingLabel(), reason,
+                    HEALTH_CHECK_PERIOD.getSeconds());
+        } else {
+            logger.debug("WebThing {} is offline {}. Try reconnect (each {} sec)", getWebThingLabel(), reason,
+                    HEALTH_CHECK_PERIOD.getSeconds());
+        }
+    }
+
+    private @Nullable URI readWebThingURI() {
+        var webThingConfiguration = getConfigAs(WebThingConfiguration.class);
+        var webThingUriString = webThingConfiguration.webThingURI;
+        if (webThingUriString == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "webThing uri has not been set");
+        } else {
+            try {
+                return URI.create(webThingUriString);
+            } catch (Exception illegalURIException) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "invalid uri " + webThingUriString);
+            }
+        }
+        return null;
+    }
+
+    private String getWebThingLabel() {
+        if (getThing().getLabel() == null) {
+            return "" + webThingURI;
+        } else {
+            return "'" + getThing().getLabel() + "' (" + webThingURI + ")";
+        }
+    }
+
+    /**
+     * updates the thing structure. Refer https://www.openhab.org/docs/developer/bindings/#updating-the-thing-structure
+     *
+     * @param webThing the WebThing that is used for the new structure
+     */
+    private void thingStructureChanged(ConsumedThing webThing) {
+        var thingBuilder = editThing().withLabel(webThing.getThingDescription().title);
```

Comment:
Labels are expected to be as short as possible. Guideline is 2-3 words with up to 25 chars. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions

Is this compatible to common WebThing devices?

Reply:
THe WebThing rfc https://iot.mozilla.org/wot/#title-member does not limit the length of the title. However, I expect that short titles will be used generally